### PR TITLE
start three ak servers simultaniously on a local machine

### PR DESCRIPTION
**Pull request by itayd**
use nginx as the load balancer. 

9980 -> nginx -> 9981,9982,9983

TODO:
- do not automigrate on start, else is flaky as several trying to setup in parallel

### CODE REVIEW

File: internal/backend/svc/svc.go [hunk 1]
```diff
@@ -298,10 +298,13 @@ func NewOpts(cfg *Config, ropts RunOptions) []fx.Option {
...
 }
 
-func StartDB(ctx context.Context, cfg *Config) (db.DB, error) {
+func StartDB(ctx context.Context, cfg *Config, ropts RunOptions) (db.DB, error) {
+	setFXRunOpts(ropts)
+
 	var db db.DB
 
 	if err := fx.New(
+		fx.NopLogger,
```
Comment (tebeka) [inline internal/backend/svc/svc.go:307]: Why `NopLogger`? We might miss some errors.

File: scripts/start-multi.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+WD="${WD-tmp/multi}"
+
+rm -fR "${WD}" && mkdir -p "${WD}"
+
+./bin/ak server setup -m dev
+
+akup() {
+    port="$1"
+    ./bin/ak up -m dev \
```
Comment (tebeka) [inline scripts/start-multi.sh:17]: Why `./bin/ak` and not `ak`? A user might want to use that without building.

File: scripts/start-multi.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (tebeka) [inline scripts/start-multi.sh:1]: Why not a docker compose file?

File: scripts/start-multi.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    echo $!
+}
+
+p1=$(akup 9981)
+p2=$(akup 9982)
+p3=$(akup 9983)
+
+trap 'kill $p1 $p2 $p3' 0
+
+nginx -c "$(pwd)/scripts/nginx.conf"
```
Comment (tebeka) [inline scripts/start-multi.sh:31]: This fails for me with:
```
./scripts/start-multi.sh 
2024/04/14 16:51:52 [emerg] 199325#199325: mkdir() "/var/lib/nginx/client-body" failed (13: Permission denied)
```

File: scripts/start-multi.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        -c "logger.zap.outputPaths=${WD}/${port}.log" \
+        >& "${WD}/${port}.out" &
+    echo $!
+}
+
+p1=$(akup 9981)
+p2=$(akup 9982)
+p3=$(akup 9983)
+
+trap 'kill $p1 $p2 $p3' 0
```
Comment (tebeka) [inline scripts/start-multi.sh:29]: When running and the hitting `CTRL-C` I get:

```
./scripts/start-multi.sh: line 1: kill: (199474) - No such process
./scripts/start-multi.sh: line 1: kill: (199476) - No such process
./scripts/start-multi.sh: line 1: kill: (199478) - No such process
```

**Review by tebeka — COMMENTED** [review]
The logs don't say which `ak` is handling a request. Should maybe add the server port to the intercept logs? Maybe give the server an ID somehow?

**Comment by efiShtain** [comment]
do you handle the multiple sqlite instances ? I didn't get where

anyhow, docker compose is the way to go here
otherwise people can decide how they want to run ak scaled

File: scripts/start-multi.sh (hunk 4 above)
Comment (ig-ra) [reply scripts/start-multi.sh:31]: I think we cannot assume that there is nginx installed.

File: scripts/start-multi.sh (hunk 5 above)
Comment (ig-ra) [reply scripts/start-multi.sh:29]: I think we shouldn't reinvent the wheel and manage subprocesses, load balancer and such.
docker compose?

**Comment by ig-ra** [comment]
> do you handle the multiple sqlite instances ? I didn't get where
> 
> anyhow, docker compose is the way to go here otherwise people can decide how they want to run ak scaled

agree with @efiShtain and @tebeka regarding docker compose

**Comment by efiShtain** [comment]
@itayd 
https://github.com/autokitteh/autokitteh/pull/186

have a look at this change
let me know if you think we should merge yours / mine / both / other options

anyhow, I don't think we should provide / promise scaling capabilities until we have the base lines of load testing

**Comment by itayd** [comment]
replaced by #186

**Pull request closed without merging**